Rakefile

```
@@ -1,50 +1,54 @@
-require 'rbconfig'
-c = RbConfig::CONFIG
-
-begin
-    require 'bundler'
-    have_bundler = true
-rescue LoadError
-    have_bundler = false
-end
+require 'mini_portile2'
+require 'rake/testtask'
 
-def system!(cmd)
-	puts cmd
-	system(cmd) or raise
+Rake::TestTask.new do |t|
+  t.libs << 'lib'
+  t.test_files = FileList['test/*_test.rb']
+  t.verbose = true
 end
 
 ver = '1.4.21'
-iter = '1'
-core = "xapian-core-#{ver}"
-bindings = "xapian-bindings-#{ver}"
-
-task :default do
-	[core,bindings].each do |x|
-		system! "tar -xJvf #{x}.tar.xz"
-	end
-
-	prefix = Dir.pwd
-
-	Dir.chdir core do
-		system! "./configure --prefix=#{prefix} --exec-prefix=#{prefix}"
-		system! "make clean all"
-		system! "make install"
-	end
-
-	Dir.chdir bindings do
-		ENV['RUBY'] ||= "#{c['bindir']}/#{c['RUBY_INSTALL_NAME']}"
-		ENV['XAPIAN_CONFIG'] = "#{prefix}/bin/xapian-config"
-		system! "./configure --prefix=#{prefix} --exec-prefix=#{prefix} --with-ruby"
-		if have_bundler
-			# Maybe Bundler is building us in a temporary directory, and will move us to
-			# the system ruby gems directory once built.
-			system! "make clean all LDFLAGS='-R#{Bundler.rubygems.gem_dir}/gems/xapian-full-alaveteli-#{ver}.#{iter}/lib'"
-		else
-			system! "make clean all"
-		end
-	end
-
-	system! "cp -RL #{bindings}/ruby/.libs/_xapian.* lib"
-	system! "cp #{bindings}/ruby/xapian.rb lib"
+package_root_dir = Dir.pwd
+
+task :core do
+  MiniPortile.new("xapian-core", ver).tap do |recipe|
+    recipe.target = File.join(package_root_dir, 'ports')
+    recipe.files << {
+      url: "https://oligarchy.co.uk/xapian/#{ver}/xapian-core-#{ver}.tar.xz",
+      sha256: '80f86034d2fb55900795481dfae681bfaa10efbe818abad3622cdc0c55e06f88'
+    }
+    recipe.patch_files = Dir[File.join(__dir__, 'patches/xapian-core', '*.patch')].sort
+    recipe.configure_options = [] # removes MiniPortile defaults
+
+    unless File.exist?(File.join(recipe.target, recipe.host, recipe.name, recipe.version))
+      recipe.cook
+    end
+
+    recipe.activate
+  end
 end
 
+task bindings: [:core] do
+  ENV['RUBY_LIB'] = ENV['RUBY_LIB_ARCH'] = File.join(__dir__, 'lib')
+
+  MiniPortile.new("xapian-bindings", ver).tap do |recipe|
+    recipe.target = File.join(package_root_dir, 'ports')
+    recipe.files << {
+      url: "https://oligarchy.co.uk/xapian/#{ver}/xapian-bindings-#{ver}.tar.xz",
+      sha256: 'e7bf1e4982b91f841289d19214021d88371ca2a6c7304df59049d7da78b83ced'
+    }
+    recipe.patch_files = Dir[File.join(__dir__, 'patches/xapian-bindings', '*.patch')].sort
+    recipe.configure_options = ['--with-ruby']
+
+    unless File.exist?(File.join(recipe.target, recipe.host, recipe.name, recipe.version))
+      recipe.cook
+    end
+
+    recipe.activate
+  end
+end
+
+desc 'Compile all extensions'
+task compile: [:bindings]
+
+task default: [:compile, :test]
```

This will require `test-unit` gem to be installed, currently it is only in the development bundle group.

```suggestion
task default: [:compile]
```